### PR TITLE
[v6r15] Allow extensions to extend DErrno

### PR DESCRIPTION
**Pull request by chaen**
Replaces https://github.com/DIRACGrid/DIRAC/pull/2571

This allows extensions to define their specific error codes
It also groups the error number by system for better tracking of them

### CODE REVIEW

File: Core/DISET/private/Transports/BaseTransport.py [hunk 1]
```diff
@@ -115,6 +114,7 @@ def _write( self, buffer ):
     return S_OK( self.oSocket.send( buffer ) )
 
   def sendData( self, uData, prefix = False ):
+    from DIRAC.Core.Utilities import DEncode
```
Comment (fstagni) [inline Core/DISET/private/Transports/BaseTransport.py]: Before we forget why, I think it's worth adding a note to this line that explains WHY it is imported here and not at the beginning of the module

File: Core/Utilities/DErrno.py [hunk 2]
```diff
@@ -10,35 +10,79 @@
 
     It also contains utilities to manipulate these errors.
 
-    Finaly, it contains a DErrno class that contains an error number
+    Finally, it contains a DErrno class that contains an error number
     as well as a low level error message. It behaves like a string for
     compatibility reasons
+
+    In order to add extension specific error, you need to create in your extension the file
+    Core/Utilities/DErrno.py, which will contains the following dictionnary:
```
Comment (fstagni) [inline Core/Utilities/DErrno.py]: will contains -> will contain
and
"dictionary" (c'est pas "dictionnaire" :) )

Anyway this should be added to the DiracDOCS. And don't forget to update https://github.com/DIRACGrid/DIRAC/wiki/DIRAC-v6r15

**Comment by fstagni** [comment]
Where includeExtensionErrors() should be added? Only to the "components" scripts as written? In theory every script may go and import code from extensions, maybe in a dynamic way, but still. So, should it be everywhere?

File: Core/DISET/private/Transports/BaseTransport.py (hunk 1 above)
Comment (chaen) [inline Core/DISET/private/Transports/BaseTransport.py]: In fact it is not necessary anymore, I move it back

**Comment by chaen** [comment]
Sorry I don't understand what you mean ? In principle, since every agent/service/executor is started with these 3 scripts, it should be safe. For the script, the includeExtensionErrors is called inside the parseCommandLine. So I guess we are covered everywhere

File: Core/Utilities/DErrno.py (hunk 2 above)
Comment (chaen) [inline Core/Utilities/DErrno.py]: PR for DiracDOCs on its way

Comment (chaen) [inline Core/Utilities/DErrno.py]: Done, and wiki updated

**Comment by fstagni** [comment]
Ok then.

**Pull request merged**